src/utils.zig

```diff
@@ -0,0 +1,31 @@
+const std = @import("std");
+
+fn bytesToChunks(
+    allocator: std.mem.Allocator,
+    bytes: []const u8,
+    chunk_size: usize
+) ![]u8 {
+    std.debug.assert(chunk_size == 1 or chunk_size == 2 or chunk_size == 4 or chunk_size == 8);
+
+    if(chunk_size == 8) return bytes;
+    
+    const chunks_per_byte = 8 / chunk_size;
+    const num_chunks = bytes.len * chunks_per_byte;
+    var chunks = try allocator.alloc(u8, num_chunks);
+    
+    var chunk_idx: usize = 0;
+    for (bytes) |byte| {
+        for (0..chunks_per_byte) |i| {
+            if (chunk_idx >= num_chunks) break;
+            
+            const shift: u8 = @intCast(i * chunk_size);
+            const mask: u8 = @intCast((1 << chunk_size) - 1);
+            chunks[chunk_idx] = (byte >> shift) & mask;
```

Comment:
for reach step you can shift the element by chunk_size rather than shifting the original element by i* chunk_size alhtough not sure if this would lead to be efficiency so may be a small test for that